test/jobs/purge_job_test.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+require 'test_case'
+
+class PurgeJobTest < ActiveSupport::TestCase
+  def setup
+    # Allowed to purge immediately for the tests
+    ADMIN_OPTIONS['job_run_email'] = 'test@test.com'
+    Patient.delete_all
+    Assessment.delete_all
+    ReportedCondition.delete_all
+    Symptom.delete_all
+  end
+
+  def teardown
+    ADMIN_OPTIONS['job_run_email'] = nil
+  end
+
+  test 'sends an email with all purged monitorees' do
+    patient = create(:patient, monitoring: false, purged: false)
+    patient.update(updated_at: (ADMIN_OPTIONS['purgeable_after'].minutes + 14.days).ago)
+    email = PurgeJob.perform_now
+    email_body = email.parts.first.body.to_s.gsub("\n", ' ')
+    assert_not ActionMailer::Base.deliveries.empty?
+    assert_includes(email_body, patient.id.to_s)
+  end
+
+  test 'sends an email with all non purged monitorees' do
+    patient = create(:patient, monitoring: false, purged: false)
+    patient.update(updated_at: (ADMIN_OPTIONS['purgeable_after'].minutes + 14.days).ago)
+
+    allow_any_instance_of(Patient).to(receive(:update!) do
+      raise StandardError, 'Test StandardError'
+    end)
+
+    email = PurgeJob.perform_now
+    email_body = email.parts.first.body.to_s.gsub("\n", ' ')
+    assert_not ActionMailer::Base.deliveries.empty?
+    assert_includes(email_body, patient.id.to_s)
+    assert_includes(email_body, 'Test StandardError')
+  end
+
+  test 'does not purge heads of household with active dependents' do
```

Comment:
Would be nice to have some more tests like this one. 
I'm thinking of tests for the following:
- making sure the appropriate attributes are nulled out
- making sure both monitoring and continuous exposure are set to false after purging
- making sure a patient who has EITHER monitoring or continuous exposure and no active dependents is also not purged

Reply:
I tend to disagree a bit 
> making sure the appropriate attributes are nulled out
These attributes are subject to change and are internal to the method. I try to test the big side effects that, if changed, would deserve some discussion such as sending the email and removing downloads, assessments, etc.

> making sure a patient who has EITHER monitoring or continuous exposure and no active dependents is also not purged
This particular case seems to be testing if the scope works or not. The scope should be tested to make sure it returns both types of patients.

Reply:
> These attributes are subject to change and are internal to the method. I try to test the big side effects that, if changed, would deserve some discussion such as sending the email and removing downloads, assessments, etc.

I understand what you mean, but we're not actually testing that the purging of identifying values is happening successfully, which is really the crux functionality of the job. Where else would we test that?

> This particular case seems to be testing if the scope works or not. The scope should be tested to make sure it returns both types of patients.

Agreed, except we're not currently checking this in the scope, we're checking it here with `next if monitoree.active_dependents.count.positive?`